pkg/mic/mic.go

```diff
@@ -58,7 +58,9 @@ type Client struct {
 	SyncLoopStarted   bool
 	syncRetryInterval time.Duration
 
-	syncing       int32 // protect against conucrrent sync's
+	syncing    int32 // protect against conucrrent sync's
+	statsMutex sync.Mutex
```

Comment:
I think this should be a stats property and locking should be internally handled by stats.

Reply:
This is really the only place where stats is updated in parallel currently. I think we can switch to moving it to stats and wrapping in structs next.